src/Builder/QueryType.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder;
+
+use MongoDB\Builder\Query\TypeOperator;
+use MongoDB\Builder\Type\Encode;
+use MongoDB\Builder\Type\FieldQueryInterface;
+use MongoDB\Builder\Type\OperatorInterface;
+
+/**
+ * Shortcut for $type field query operator
+ *
+ * @see https://www.mongodb.com/docs/manual/reference/operator/query/type/#available-types
+ * @see TypeOperator
+ */
+enum QueryType: string implements OperatorInterface, FieldQueryInterface
+{
+    public const ENCODE = Encode::Array;
```

Comment:
Is this the first time a BackedEnum is being used for an encoding type? If so, perhaps it'd be prudent to have `BuilderEncoder::encode()` assert that that `$value::ENCODE` is `Encode::Array` (and you can revise that as needed should another use case come up).

That would address my question about how a BackedEnum might be handled for other `switch` cases.

Reply:
We need to rework the while encoder class. The "ENCODE" constant will not stay, it may become a method of the Operator interface.